source/server/admin/admin.h

```diff
@@ -68,6 +68,7 @@ class AdminImpl : public Admin,
                   public Network::FilterChainFactory,
                   public Http::FilterChainFactory,
                   public Http::ConnectionManagerConfig,
+                  public Server::Chunker,
```

Comment:
Why do you think it's helpful to have AdminImpl inherit from Chunker?

Reply:
I thought that admin should use `Chunker` instead of `Instance`, so inherit.

> Looks like this is still in progress. Can you ping when you have a working implementation of Chunker? Even if it just delegates to Buffer::Instance and doesn't actually do any chunking, this might be worth putting into the codebase as an intermediate point.

From this, I get it, so that I just need to implement `reportError` and re-define `MAKE_ADMIN_HANDLER` to remove http code return, is anything wrong?

Reply:
There's only one instance of Admin, but. there can ultimately I think be multiple Chunkers all asynchronously generating content for concurrent async admin endpoints.

Also look at it this way: we were using Buffer::Instance and replacing it with Chunker. We were not previously inheriting from Buffer::Instance so why do you think it helps to inherit from Chunker?

Reply:
Oh, it is helpless I see.